source/Calamari.AzureAppService/Behaviors/AzureAppServiceBehaviour.cs

```diff
@@ -177,22 +182,45 @@ private async Task<IDeploymentSlot> FindOrCreateSlot(IWebApp client, TargetSite
             var zipUploadUrl = $"{publishingProfile.PublishUrl}{Archive.UploadUrlPath}";
             Log.Verbose($@"Publishing {uploadZipPath} to {zipUploadUrl}");
 
-            // The HttpClient default timeout is 100 seconds: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-5.0#remarks
-            // This timeouts with even relatively small packages: https://octopus.zendesk.com/agent/tickets/69928
-            // We'll set this to an hour for now, but we should probably implement some more advanced retry logic, similar to https://github.com/OctopusDeploy/Sashimi.AzureWebApp/blob/bbea36152b2fb531c2893efedf0330a06ae0cef0/source/Calamari/AzureWebAppBehaviour.cs#L70
-            client.Timeout = TimeSpan.FromHours(1);
+            using var httpClient = new HttpClient(new HttpClientHandler
+            {
+#pragma warning disable DE0003
+                Proxy = WebRequest.DefaultWebProxy
+#pragma warning restore DE0003
+            })
+            {
+                // The HttpClient default timeout is 100 seconds: https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-5.0#remarks
+                // This timeouts with even relatively small packages: https://octopus.zendesk.com/agent/tickets/69928
+                // We'll set this to an hour for now, but we should probably implement some more advanced retry logic, similar to https://github.com/OctopusDeploy/Sashimi.AzureWebApp/blob/bbea36152b2fb531c2893efedf0330a06ae0cef0/source/Calamari/AzureWebAppBehaviour.cs#L70
+                Timeout = TimeSpan.FromHours(1)
+            };
 
-            var response = await client.PostAsync($@"{zipUploadUrl}",
-                new StreamContent(new FileStream(uploadZipPath, FileMode.Open, FileAccess.Read, FileShare.Read))
-                {
-                    Headers = { ContentType = new MediaTypeHeaderValue("application/octet-stream") }
-                }
-            );
 
-            if (!response.IsSuccessStatusCode) 
-                throw new Exception($"Zip upload to {zipUploadUrl} failed with HTTP Status '{response.StatusCode} {response.ReasonPhrase}'.");
+            //we add some retry just in case the web app's Kudu/SCM is not running just yet
+            var response = await RetryPolicies.TransientHttpErrorsPolicy.ExecuteAsync(async () =>
```

Comment:
I don't _love_ adding this retry here but I was getting failures in the tests as the SCM/Kudu site wasn't ready before the test ran. I guess we could adjust the test to do some availability checks before executing the test?

Happy to take feedback on this one

Reply:
Hrm I would probably just leave it as you've done it with the retires. As long as the tests are stable like that.